elastic/beats
# [add_cloud_metadata] Remove logger for AWS/EC2

## What

Remove the logger for the provider AWS/EC2 in `add_cloud_metadata`.


## Details

`add_cloud_metadata` is enabled by default. When the provider AWS/EC2 is setup we see this message at all times, even when we are not using AWS/EC2:

```
{"log.level":"warn","@timestamp":"2023-07-06T18:24:06.085Z","message":"error fetching EC2 Identity Document: operation error ec2imds: GetInstanceIdentityDocument, canceled, context deadline exceeded.","component":{"binary":"filebeat","dataset":"elastic_agent.filebeat","id":"filestream-monitoring","type":"filestream"},"log":{"source":"filestream-monitoring"},"log.logger":"add_cloud_metadata","log.origin":{"file.line":91,"file.name":"add_cloud_metadata/provider_aws_ec2.go"},"service.name":"filebeat","ecs.version":"1.6.0","ecs.version":"1.6.0"}
```

This change was introduced by this PR: https://github.com/elastic/beats/pull/28285.

It is happening because when creating the provider, we call this function:

https://github.com/elastic/beats/blob/e322104a8cc25b420f7764a50a9f801e7cba6aa0/libbeat/processors/add_cloud_metadata/add_cloud_metadata.go#L73

And then we try to create each provider of the list:

https://github.com/elastic/beats/blob/e322104a8cc25b420f7764a50a9f801e7cba6aa0/libbeat/processors/add_cloud_metadata/providers.go#L115

The problem comes when this function is called. We have a logger that will start outputting warnings:

https://github.com/elastic/beats/blob/e322104a8cc25b420f7764a50a9f801e7cba6aa0/libbeat/processors/add_cloud_metadata/provider_aws_ec2.go#L78

If we check the other providers' files (`provider_*.go`), we can check that none of them are using loggers inside their `Create`.

The easiest solution for now is just to delete the loggers.


## Checklist




## How to test this PR locally

The easiest way is to:
1. Clone this branch.
2. Build metricbeat.
3. Start it.
4. Check the error is still no longer there.


## Related issues

- Closes https://github.com/elastic/beats/issues/36018.
- Relates to https://github.com/elastic/obs-infraobs-team/issues/1204.



## Logs

It is no longer present in the logs.
